pkg/segment/writer/segmetarw.go

```diff
@@ -480,12 +492,107 @@ func BackFillPQSSegmetaEntry(segkey string, newpqid string) {
 	if sfmData.AllPQIDs == nil {
 		sfmData.AllPQIDs = make(map[string]bool)
 	}
-	sfmData.AllPQIDs[newpqid] = true
 
-	// TODO this should be through a channel so that we only have one writer
+	utils.MergeMapsRetainingFirst(sfmData.AllPQIDs, pqidMap)
+
 	writeSfm(segkey, sfmData)
 }
 
+func BackFillPQSSegmetaEntry(segkey string, newpqid string) {
+	BulkBackFillPQSSegmetaEntries(segkey, map[string]bool{newpqid: true})
+}
+
+// AddToBackFillAndEmptyPQSChan adds a new pqid to the channel
+// if emptyPqs is true, then it will also add the EmptyResults for this pqid
+func AddToBackFillAndEmptyPQSChan(segkey string, newpqid string, emptyPqs bool) {
+	pqsChan <- PQSChanMeta{pqid: newpqid, segKey: segkey, emptyPqs: emptyPqs}
+}
+
+func listenBackFillAndEmptyPQSRequests() {
+	// Listen on the channel, every 10 seconds or if the size of the channel is 100,
+	// it would get all the data in the channel and then do the process of Backfilling PQMR files.
+	// This is to avoid multiple writes to the same file
+
+	ticker := time.NewTicker(10 * time.Second) // every 10 seconds
+	defer ticker.Stop()
+
+	buffer := make([]PQSChanMeta, 0, 100)
```

Comment:
why is this called Meta ?

Reply:
I named it Meta because `segkey` and `pqid`, we are adding this information segment Meta and pqid empty meta.